packages/headless/src/features/commerce/configuration/configuration-actions.ts

```diff
@@ -0,0 +1,44 @@
+import {createAction} from '@reduxjs/toolkit';
+import {nonEmptyString, validatePayload} from '../../../utils/validate-payload';
+import {
+  UpdateAnalyticsConfigurationActionCreatorPayload,
+  UpdateBasicConfigurationActionCreatorPayload,
+  analyticsConfigurationSchema,
+} from '../../configuration/configuration-actions';
+
+export type UpdateBasicConfigurationPayload =
+  UpdateBasicConfigurationActionCreatorPayload;
+
+export const updateBasicConfiguration = createAction(
+  'commerce/configuration/updateBasicConfiguration',
+  (payload: UpdateBasicConfigurationPayload) =>
+    validatePayload(payload, {
+      accessToken: nonEmptyString,
+      organizationId: nonEmptyString,
+      platformUrl: nonEmptyString,
+    })
+);
+
+export type UpdateAnalyticsConfigurationPayload = Pick<
```

Comment:
This is a significant change, here's my reasoning:

- `originContext`, `originLevel2`, and `originLevel3` are not needed beacause they are not leveraged by the commerce API (i.e., server-side events) or by Relay (i.e., client-side events), so they don't seem relevant to me. Arguably, `originLevel3` would normally be the referrer, but that's retrieved through the navigator context provider.
- `apiBaseUrl` / `nextApiBaseUrl` is not needed because it would be the same as for search requests (for server-side events), or it would be managed by Relay (for client-side events). I might be wrong about that though.
- I suppose `runtimeEnvironment` should be inferred (or explicitely specified) in the navigator context provider.
- `anonymous` not supported by the commerce API or Relay AFAIK
- `deviceId` I believe this is not useful because the clientId is sufficient
- `userDisplayName` not supported by commerce API or Relay AFAIK.
- `documentLocation` this would be the view.url in the context of commerce.
- `analyticsMode` only next is supported in commerce

Bear in mind I may very well be wrong about any of these, so please challenge this ;)

Reply:
I'm hesitant mostly on the following properties:

- `nextApiBaseUrl`
- `runtimeEnvironment`

@louis-bompart, a penny for your thoughts 😄

Reply:
As discussed in, worst case scenario we can always add some properties back.